src/common/blkdev.h

```diff
@@ -1,29 +1,48 @@
 #ifndef __CEPH_COMMON_BLKDEV_H
 #define __CEPH_COMMON_BLKDEV_H
 
-/* for testing purposes */
-extern void set_block_device_sandbox_dir(const char *dir);
-
-// from a path (e.g., "/dev/sdb")
-extern int get_block_device_base(const char *path, char *devname, size_t len);
-
-// from an fd
-extern int block_device_discard(int fd, int64_t offset, int64_t len);
-extern int get_block_device_size(int fd, int64_t *psize);
-extern int get_device_by_fd(int fd, char* partition, char* device, size_t max);
-
-// from a uuid
-extern int get_device_by_uuid(uuid_d dev_uuid, const char* label,
-			      char* partition, char* device);
-
-// from a device (e.g., "sdb")
-extern int64_t get_block_device_int_property(
-	const char *devname, const char *property);
-extern int64_t get_block_device_string_property(
-	const char *devname, const char *property,
-	char *val, size_t maxlen);
-extern bool block_device_support_discard(const char *devname);
-extern bool block_device_is_rotational(const char *devname);
-extern int block_device_model(const char *devname, char *model, size_t max);
+#include <utility>
+
+enum blkdev_prop_t {
+  BLKDEV_PROP_DEV,
+  BLKDEV_PROP_DISCARD_GRANULARITY,
+  BLKDEV_PROP_MODEL,
+  BLKDEV_PROP_ROTATIONAL,
+  BLKDEV_PROP_SERIAL,
+  BLKDEV_PROP_VENDOR,
+  BLKDEV_PROP_NUMPROPS
+};
+
+class BlkDev {
+public:
+  BlkDev(int fd);
+
+  /* GoogleMock requires a virtual destructor */
+  virtual ~BlkDev() {}
+
+  // from an fd
+  int block_device_discard(int64_t offset, int64_t len);
+  int get_block_device_size(int64_t *psize);
+  std::pair<int, dev_t> get_devid();
```

Comment:
@asomers nit: why not `int get_devid(dev_t *)`? I have nothing against returning it as a pair in general, but this just looks inconsistent with other functions here.

Reply:
@trociny I changed get_devid to return a pair because I thought that I would need to mock it with googlemock, and it's easier to mock return values than return arguments.  It turns out that this wasn't the best place to mock, though. But I left the signature this way when I realized that it's probably more efficient than using a return argument since the pair can fit into a single 64-bit register.  Do you want me to change it back?

Reply:
I don't think we have to worry about efficiency here, and the code readability is more important. From this point of view I would prefer all getters follow the same style: either all return only the error code, or all return a pair (the error code, value).

But this is really a bikeshedding. I don't mind if it stays as is.